core/operation/src/main/java/uk/gov/gchq/gaffer/jobtracker/Job.java

```diff
@@ -61,32 +53,48 @@ public void setRepeat(final Repeat repeat) {
         this.repeat = repeat;
     }
 
-    public String getOpChain() {
-        try {
-            if (opChain instanceof OperationChainDAO) {
-                return new String(JSONSerialiser.serialise(opChain),
-                        Charset.forName(CHARSET_NAME));
-            } else {
-                final OperationChainDAO dao = new OperationChainDAO(opChain.getOperations());
-                return new String(JSONSerialiser.serialise(dao),
-                        Charset.forName(CHARSET_NAME));
-            }
-        } catch (final SerialisationException se) {
-            throw new IllegalArgumentException(se.getMessage());
-        }
+    public Operation getOperation() {
+        return operation;
     }
 
-    @JsonIgnore
-    public OperationChain<?> getOpChainAsOperationChain() {
-        return opChain;
+    public void setOperation(final Operation operation) {
+        this.operation = operation;
     }
 
-    public void setOpChain(final String opChain) {
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (null == obj || getClass() != obj.getClass()) {
+            return false;
+        }
+        final Job job = (Job) obj;
+
         try {
-            this.opChain = JSONSerialiser.deserialise(opChain,
-                    OperationChainDAO.class);
+            return new EqualsBuilder()
+                    .append(JSONSerialiser.serialise(operation), JSONSerialiser.serialise(job.operation))
```

Comment:
Is this being used because not all operations implement equals method?

Reply:
If so see gh-2157

Reply:
In short, yes. There are quite a few operations which don't have Equals/Hashcode method so in order to compare them, I used the JSON serialised form since all operations should be json serialisable.